manifests/params.pp

```diff
@@ -17,7 +22,7 @@
           $cert_dir_mode     = '2665'
         }
         default: {
-          fail("Unsupported operatingsystem (${facts['os']['name']})")
+          $cert_dir_mode     = '0755'
```

Comment:
I remove all "Unsupported os" fails as that's hard to fit when moving to hiera.
Given that all settings here are now class parameters I really think the module supports any Linux OS out there.